src/ntt/mod.rs

```diff
@@ -19,56 +17,111 @@ pub use self::{
     wavelet::{inverse_wavelet_transform, wavelet_transform},
 };
 
-/// RS encode at a rate 1/`expansion`.
-#[cfg_attr(feature = "tracing", instrument(skip(coeffs), fields(size = coeffs.len())))]
-pub fn expand_from_coeff<F: FftField>(coeffs: &[F], expansion: usize) -> Vec<F> {
-    let engine = cooley_tukey::NttEngine::<F>::new_from_cache();
-    let expanded_size = coeffs.len() * expansion;
-    let mut result = Vec::with_capacity(expanded_size);
-    // Note: We can also zero-extend the coefficients and do a larger NTT.
-    // But this is more efficient.
-
-    // Do coset NTT.
-    let root = engine.root(expanded_size);
-    result.extend_from_slice(coeffs);
-    #[cfg(not(feature = "parallel"))]
-    for i in 1..expansion {
-        let root = root.pow([i as u64]);
-        let mut offset = F::ONE;
-        result.extend(coeffs.iter().map(|x| {
-            let val = *x * offset;
-            offset *= root;
-            val
-        }));
-    }
-    #[cfg(feature = "parallel")]
-    result.par_extend((1..expansion).into_par_iter().flat_map(|i| {
-        let root_i = root.pow([i as u64]);
-        coeffs
-            .par_iter()
-            .enumerate()
-            .map_with(F::ZERO, move |root_j, (j, coeff)| {
-                if root_j.is_zero() {
-                    *root_j = root_i.pow([j as u64]);
-                } else {
-                    *root_j *= root_i;
-                }
-                *coeff * *root_j
-            })
-    }));
-
-    ntt_batch(&mut result, coeffs.len());
-    transpose(&mut result, expansion, coeffs.len());
+///
+/// RS encode a polynomial `poly` of degree `d-1` at the rate
+/// 1/`expansion` (i.e., the RS block length `N = d*expansion`).
+///
+/// Suppose the Merkle tree leaves are grouped as cosets of size  k =
+/// 2^`fold_factor`. Given a polynomial f(x) of degree `d-1`, it can be
+/// extended by zero-padding to a polynomial of degree `N-1` and written
+/// in the following form:
+///
+/// f(X) = h₀(X^k) + X·h₁(X^k) + ⋯ + X^{k-1}·h_{k-1}(X^k)
+///
+/// where each hₜ(X) is polynomial of maximum degree `N/k - 1`.
+///
+/// If ζ is the N-th root of unity, then [rs_encode_coset_poly] computes
+/// the Reed-Solomon code for each hₜ(x) using `ω = ζ^k` as the root of
+/// unity, and for each index i, places [h₀(ω^i), h₁(ω^i), ⋯
+/// ,h_{k-1}(ω^i)] consecutively in the output buffer.
+///
```

Comment:
Alternatively, a better way of thinking about this (I think) is that this is encoding a message f = (f_1, ..., f_n) as the interleaved Reed solomon codeword 
i.e. you take (f_1, f_k, f_2k, ...), (f_2, ...) and you encode each one of them with a RS code of size n/k.
Maybe we should update this comment and the naming of the function appropriately (how about `interleaved_rs_encode`?)

Let's also maybe leave a TODO on whether it makes sense to store the polynomial already transposed (i.e. poly as (f_1, f_k, f_2k,..., f_2, f_2 +k, ...))
This might save us some of the transposes? Not sure if in other places this helps or hurts